controllers/eventtrigger_deployer.go

```diff
@@ -886,7 +888,12 @@ func instantiateClusterProfileForResource(ctx context.Context, c client.Client,
 	if resource != nil {
 		object.Resource = resource.UnstructuredContent()
 	}
-
+	cluster, err := fecthClusterObjects(ctx, c, clusterNamespace, clusterName, clusterType, logger)
```

Comment:
Thank you. I feel since we are on this path of extending templating, we can also enhance it so that the name of the referenced ConfigMap/Secret instances can be expressed as a template and instantiate using Cluster namespace/name.

We need to change this https://github.com/projectsveltos/event-manager/blob/18856d0216f814edf753e014c5954071170fe03b/controllers/fetcher.go#L128 

Instead of using `Name: policyRef.Name` we use this 

https://github.com/projectsveltos/libsveltos/blob/75fbc59d75dc0a81aa57202542cc3711e416b084/lib/template/referenced_resource.go#L45

what do you think? 

And thank you again for this. I really like it.

Reply:
If I get this correctly you would like to extend the fetch objection, in  the way that a user can inject Template like that ?: 

```yaml
...
  policyRefs:
  - name: loadbalancer-class-handler
    namespace: "{{ .Cluster.metdata.namespace }}"
    kind: ConfigMap
 ...   
 ```

Reply:
yes that is correct. One clarification though. To be consistent:

1.  namespace can either be specified (with no template) or left empty (and Sveltos will use the cluster namespace). This is already in place if you look at https://github.com/projectsveltos/event-manager/blob/18856d0216f814edf753e014c5954071170fe03b/controllers/fetcher.go#L126
2. the name instead can be expressed as a template. For instance my-configmap-{{ .Cluster.metadata.name }} and that would be useful in the case where we want to fetch different configMaps per cluster 

BTW, we don't have to put both changes in a single PR. What you have is already good enough. It just inspired me to extend the templating with event framework a little more

Reply:
ok got u